sc-memory/sc-memory/sc_scs_helper.cpp

```diff
@@ -267,12 +270,24 @@ class StructGenerator
     }
   }
 
+  void GenerateFormatInfo(ScAddr const & addr, std::string const & fileName)
+  {
+    int pos = fileName.find_last_of('.');
+    std::string const fmtStr = "format_" + fileName.substr(pos + 1);
```

Comment:
Crash there. What if '.' is not found.

Reply:
What if URL is and `http://....` and there are no extension or some query items exists?